imports/client/components/PuzzleListPage.tsx

```diff
@@ -385,53 +417,45 @@ const PuzzleListView = ({
-                <ToggleButton id="view-unlock-button" variant="outline-info" value="unlock">Unlock</ToggleButton>
-              </ToggleButtonGroup>
-              <ToggleButtonGroup
-                type="checkbox"
-                value={showSolved ? ['true'] : []}
-                onChange={changeShowSolved}
-              >
-                <ToggleButton id="view-show-solved-button" variant="outline-info" value="true">Show solved</ToggleButton>
-              </ToggleButtonGroup>
-            </ButtonToolbar>
-          </ViewControlsSection>
-          <ViewControlsSectionExpand>
-            <FormLabel htmlFor="jr-puzzle-search">
-              {`Showing ${retainedPuzzles.length}/${allPuzzles.length} items`}
-            </FormLabel>
-            <FilterToolbar>
-              <FilterToolbarInputGroup>
-                <FormControl
-                  id="jr-puzzle-search"
-                  as="input"
-                  type="text"
-                  ref={searchBarRef}
-                  placeholder="Filter by title, answer, or tag"
-                  value={searchString}
-                  onChange={onSearchStringChange}
-                />
-                <Button variant="secondary" onClick={clearSearch}>
-                  <FontAwesomeIcon icon={faEraser} />
-                </Button>
-              </FilterToolbarInputGroup>
-            </FilterToolbar>
-          </ViewControlsSectionExpand>
-          <ViewControlsSection>
-            <ButtonToolbar>
-              {addPuzzleContent}
-            </ButtonToolbar>
-          </ViewControlsSection>
-        </ViewControls>
-      </FormGroup>
-      {renderList(retainedPuzzles, solvedOverConstrains)}
+      <ViewControls canAdd={canAdd}>
+        <FormGroup>
+          <FormLabel>Organize by</FormLabel>
+          <ButtonToolbar>
+            <StyledToggleButtonGroup type="radio" name="puzzle-view" defaultValue="group" value={displayMode} onChange={setDisplayMode}>
+              <ToggleButton id="view-group-button" variant="outline-info" value="group">Group</ToggleButton>
+              <ToggleButton id="view-unlock-button" variant="outline-info" value="unlock">Unlock</ToggleButton>
+            </StyledToggleButtonGroup>
+          </ButtonToolbar>
+        </FormGroup>
+        <FormGroup>
+          <FormLabel>Solved puzzles</FormLabel>
+          <ButtonToolbar>
+            <StyledToggleButtonGroup type="radio" name="show-solved" defaultValue="show" value={showSolved ? 'show' : 'hide'} onChange={setShowSolvedString}>
+              <ToggleButton id="solved-hide-button" variant="outline-info" value="hide">Hidden</ToggleButton>
+              <ToggleButton id="solved-show-button" variant="outline-info" value="show">Shown</ToggleButton>
+            </StyledToggleButtonGroup>
+          </ButtonToolbar>
+        </FormGroup>
+        {addPuzzleContent}
+        <SearchFormGroup canAdd={canAdd}>
+          <SearchFormLabel canAdd={canAdd}>Search</SearchFormLabel>
```

Comment:
I'm getting a React warning about `label` not expecting a `canAdd` prop:

> Warning: React does not recognize the `canAdd` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `canadd` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

This can be solved by prefixing the prop name with `$`, which tells styled-components not to pass through the prop to the component itself.

Reply:
https://styled-components.com/docs/api#transient-props

Reply:
Good catch. Should be fixed now.